recommendations/local-resource-conservation.md

```diff
@@ -0,0 +1,333 @@
+target route. This attack can exhaust the resources along the target route in 
+two ways, though the difference between the two behaviors is not rigorously 
+defined:
+* Quick Jamming: sending a continuous stream of payments through a route that 
+  are quickly resolved, but block all of the liquidity or HTLC slots along the 
+  route.
+* Slow Jamming: sending a slow stream of payments through a route and only 
+  failing them at the latest possible timeout, blocking all of the liquidity 
+  or HTLC slots along the route for the duration.
+
+Fees are currently only paid to routing nodes on successful completion of a 
+payment, so this attack is virtually free - the attacker pays only for the 
+costs of running a node, funding channels on-chain and the opportunity cost of 
+capital committed to the attack.
+
+This document outlines recommendations for implementing local resource 
+conservation to mitigate slow jamming attacks. It is part of a hybrid solution 
+to mitigating channel jamming, intended to be followed by the introduction of 
+unconditional fees to mitigate fast jamming.
+
+## Overview
+Local resource conservation combines the follow elements to provide nodes with 
+a mechanism to protect themselves against slow-jamming attacks:
+
+* [HTLC Endorsement](#../02-peer-protocol.md#adding-an-htlc-update_add_htlc): 
+  propagation of a signal along the payment route that indicates whether the 
+  node sending `update_add_htlc` recommends that the HTLC should be granted 
+  access to downstream resources (and that it will stake its reputation on the 
+  fast resolution of the HTLC).
+* [Local Reputation](#local-reputation): local tracking of the historical 
+  forwarding behavior of channel peers, used to decide whether to grant 
+  incoming HTLCs full access to local resources and whether to propagate 
+  endorsement downstream.
+* [Resource Bucketing](#resource-bucketing): reservation of a portion of 
+  "protected" liquidity and slots for endorsed HTLCs that have been forwarded 
+  by high reputation nodes. 
+
+Sequence: 
+* The `update_add_hltc` is sent by an upstream peer.
+* If the sending peer has sufficient local reputation (per the receiving 
+  peer's view) AND the incoming HTLC was sent with `endorsed` set to `1`: 
+  * The HTLC will be allowed access to the "protected" bucket, so will have 
+    full access to the channel's liquidity and slots, 
+  * The corresponding outgoing HTLC (if present) will be forwarded with 
+    `endorsed` set to `1`.
+* Otherwise: 
+  * The HTLC will be limited to the remaining "general" slots and liquidity, 
+    and will be failed if there are no resources remaining in this bucket.
+  * The corresponding outgoing HTLC (if present) will be forwarded with 
+    `endorsed` set to `0`.
+
+In the steady state when Lightning is being used as expected, there should be 
+no need for use of the protected slots as payments are resolved quickly. Should 
```

Comment:
> as payments are resolved quickly

Maybe as the full capacity of slots and liquidity is rarely used.

Reply:
slots/resources 
+ do not saturate the channel